### ISSUE TITLE
Invalid date in body results in a 500 error

### ISSUE THREAD

**Issue by duncanportelli** [crnk-project/crnk-framework#437]
When passing an invalid date in the request body (e.g. `"birthDate" : "1992-05-32"`) a `ResourceException` is thrown ([`ResourceUpsert.java:154`](https://github.com/crnk-project/crnk-framework/blob/master/crnk-core/src/main/java/io/crnk/core/engine/internal/dispatcher/controller/ResourceUpsert.java#L178)) because the date cannot be deserialized to a LocalDate. This is handled as a `500 Internal Server` error. 

Shouldn't this be a `400 Bad Request` error?

**Comment by remmeier** [comment]
yes, it should, at least a 4xx. PRs always welcome should you already have a fix in mind.

**Comment by weaselmetal** [comment]
A crnk built-in `ExceptionMapper` that catches (certain) `JsonProcessingException`s should do the trick, no?
